rest-servlet/src/main/java/org/hawkular/metrics/restServlet/influx/InfluxSeriesHandler.java

```diff
@@ -435,11 +437,65 @@ private String getColumnName(SelectQueryDefinitions queryDefinitions) {
                     NumberFunctionArgument argument = (NumberFunctionArgument) aggregationFunctionArguments.get(1);
                     retVal = quantil(list, argument.getDoubleValue());
                     break;
+                case TOP:
+                    isMultipleValues = true;
+                    if (!list.isEmpty() && aggregationFunctionArguments.size() > 0) {
+                        int numberOfTopElement = list.size() >
```

Comment:
You shouldn't check the argument list here. If you attach a validator when declaring the function, the parsing tool will check the arguments for you.